src/gov-action-contracts/gov-upgrade-contracts/upgrade-proxy/MultiProxyUpgradeAction.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.16;
+
+import {TimelockControllerUpgradeable} from
+    "@openzeppelin/contracts-upgradeable/governance/TimelockControllerUpgradeable.sol";
+import {ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {TransparentUpgradeableProxy} from
+    "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {ProxyUpgradeAction} from "./ProxyUpgradeAction.sol";
+
+/// @title MultiProxyUpgradeAction
+/// @notice A contract to proxy upgrade the Core and Treasury Governor contracts.
+/// @custom:security-contact https://immunefi.com/bug-bounty/arbitrum/information/
+contract MultiProxyUpgradeAction is ProxyUpgradeAction {
+    /// @notice The address of the Proxy Admin contract.
+    address public immutable PROXY_ADMIN;
+    /// @notice The address of the current Core Governor contract.
+    address public immutable CURRENT_CORE_GOVERNOR;
+    /// @notice The address of the new Core Governor contract.
+    address public immutable NEW_CORE_GOVERNOR_IMPLEMENTATION;
+    /// @notice The address of the current Treasury Governor contract.
+    address public immutable CURRENT_TREASURY_GOVERNOR;
+    /// @notice The address of the new Treasury Governor contract.
+    address public immutable NEW_TREASURY_GOVERNOR_IMPLEMENTATION;
+
+    /// @notice Sets up the contract with the given parameters.
+    /// @param _proxyAdmin The address of the Proxy Admin contract.
+    /// @param _currentCoreGovernor The address of the current Core Governor contract.
+    /// @param _newCoreGovernor The address of the new Core Governor contract.
+    /// @param _currentTreasuryGovernor The address of the current Treasury Governor contract.
+    /// @param _newTreasuryGovernor The address of the new Treasury Governor contract.
+    constructor(
+        address _proxyAdmin,
+        address _currentCoreGovernor,
+        address _newCoreGovernor,
```

Comment:
i think we need different var names. also, the natspec should be updated. we're talking here not about "new core governor" but instead "new governor implementation." and this is the same for both core and treasury, so we probably only need 1 var for the impl itself.